godal.go

```diff
@@ -3849,6 +3849,45 @@ func (ds *Dataset) Grid(destPath string, switches []string, opts ...GridOption)
 	return &Dataset{majorObject{C.GDALMajorObjectH(dsRet)}}, nil
 }
 
+// Grid runs the library version of nearblack.
+// See the nearblack doc page to determine the valid flags/opts that can be set in switches.
+//
+// Example switches :
+//
+//	[]string{"-white", "-near", "10"}
+//
+// Creation options and driver may be set in the switches slice with
+//
+//	switches:=[]string{"-co","TILED=YES","-of","GTiff"}
+//
+// NOTE: Some switches are NOT compatible with this binding, as a `nullptr` is passed to a later call to
+// `GDALNearblackOptionsNew()` (as the 2nd argument). Those switches are: "-o", "-q", "-quiet"
+func (ds *Dataset) Nearblack(destPath string, destDs *Dataset, switches []string, opts ...NearblackOption) (*Dataset, error) {
```

Comment:
Please split this into 2 functions, in the same way that is done for `Warp` or `Rasterize`, making clear that destPath ans destDs are mutually exclusive:

- func (ds *Dataset) Nearblack(destPath string, switches []string, opts ...NearblackOption) (*Dataset, error)
- func (ds *Dataset) NearblackInto(srcDs *Dataset, switches []string, opts ...NearblackOption) error